scylla/src/transport/cluster.rs

```diff
@@ -339,19 +362,34 @@ impl ClusterWorker {
                 recv_res = self.server_events_channel.recv() => {
                     if let Some(event) = recv_res {
                         debug!("Received server event: {:?}", event);
-                        match event {
-                            Event::TopologyChange(_) => (), // Refresh immediately
+
+                        let mut is_refresh_needed = false;
+
+                        match &event {
+                            Event::TopologyChange(_) => {
+                                // Refresh immediately
+                                is_refresh_needed = true;
+                            },
                             Event::StatusChange(status) => {
                                 // If some node went down/up, update it's marker and refresh
                                 // later as planned.
 
                                 match status {
-                                    StatusChangeEvent::Down(addr) => self.change_node_down_marker(addr, true),
-                                    StatusChangeEvent::Up(addr) => self.change_node_down_marker(addr, false),
+                                    StatusChangeEvent::Down(addr) => self.change_node_up_marker(*addr, false),
+                                    StatusChangeEvent::Up(addr) => self.change_node_up_marker(*addr, true),
                                 }
-                                continue;
                             },
-                            _ => continue, // Don't go to refreshing
+                            Event::SchemaChange(_) => {},
+                        }
+
+                        let cluster_data = self.cluster_data.load_full();
+                        for consumer in self.event_consumers.iter() {
+                            consumer.consume(&[event.clone()], &cluster_data);
```

Comment:
Tip: no need to clone, you can convert a reference to a single-element slice in Rust. See https://doc.rust-lang.org/stable/std/slice/fn.from_ref.html